src/Mooncake.jl

```diff
@@ -149,7 +149,7 @@ include("public.jl")
 end
 #! format: on
 
-@public Config, value_and_pullback!!, prepare_pullback_cache
+@public Config, value_and_pullback!!, prepare_pullback_cache, require_tangent_cache
```

Comment:
As discussed when we met, no need to export this function. Let's also adopt the simpler design, i.e. `prepare_tangent_cache`. 

```suggestion
@public Config, value_and_pullback!!, prepare_pullback_cache
```